### PR TITLE
Implement promotion UI and handle promotion

### LINKED ISSUE #56: Support pawn promotion
We do not have pawn promotion implemented yet.

This needs a logic for promotion and the UI for user to pick up what he wants the pawn to be promoted to.

### PR DESCRIPTION
**Pull request by Prabhakar-Poudel**
closes: #56

### CODE REVIEW

**Comment by nimish-gupta** [comment]
add screenshots

**Comment by Prabhakar-Poudel** [comment]
GIFs:

whitepromotion



blackpromotion

**Comment by Prabhakar-Poudel** [comment]
@nimish-gupta Added GIFs.

File: client/src/components/chess-board/PawnPromotion.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChessSquare from '../chess-square/ChessSquare';
+import './PawnPromotion.css';
+
+function PawnPromotion({ onSelect, pieceColor }) {
+  const pieceTypes = ['r', 'n', 'b', 'q'];
```
Comment (sagarbakhtar) [inline client/src/components/chess-board/PawnPromotion.js]: We should put this in the config file.

File: client/src/components/chess-board/ChessBoard.test.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ describe('<ChessBoard />', () => {
         squares={squares}
         board={board}
         movePiece={() => true}
+        playerColor="w"
```
Comment (sagarbakhtar) [inline client/src/components/chess-board/ChessBoard.test.js]: Can use constants from config file

File: client/src/components/chess-board/PawnPromotion.js (hunk 1 above)
Comment (sagarbakhtar) [inline client/src/components/chess-board/PawnPromotion.js]: We should put this in the config file.

File: client/src/components/chess-board/ChessBoard.test.js (hunk 2 above)
Comment (sagarbakhtar) [inline client/src/components/chess-board/ChessBoard.test.js]: Can use constants from config file

**Review by sagarbakhtar — CHANGES_REQUESTED** [review]

**Review by sagarbakhtar — APPROVED** [review]

**Pull request merged**